test/unit/cloudwatch_agent/test_write_cloudwatch_agent_json.py

```diff
@@ -0,0 +1,194 @@
+        "feature_conditions": [],
+    },
+]
+
+METRIC_CONFIGS = {
+    "metrics_collected": [
+        {
+            "metric_type": "mem",
+            "measurement": ["mem_used_percent"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+        {
+            "metric_type": "disk",
+            "measurement": ["disk_used_percent"],
+            "resources": ["/"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+    ],
+    "append_dimensions": ["InstanceId"],
+    "aggregation_dimensions": [["InstanceId", "path"]],
+}
+
+
+def test_add_log_group_name_params():
+    configs = add_log_group_name_params("test", CONFIGS)
+    for config in configs:
+        assert_that(config).contains("log_group_name")
+        assert_that(config["log_group_name"]).is_equal_to("test")
+
+
+def test_add_instance_log_stream_prefixes():
+    configs = add_instance_log_stream_prefixes(CONFIGS)
+    for config in configs:
+        assert_that(config["log_stream_name"]).contains(gethostname())
+
+
+@pytest.mark.parametrize(
+    "dimensions",
+    [
+        {"platform": "amazon", "length": 2},
+        {"platform": "ubuntu", "length": 2},
+        {"scheduler": "slurm", "length": 2},
+        {"scheduler": "awsbatch", "length": 3},
+    ],
+)
+def test_select_configs_for_dimesion(dimensions):
+    if "platform" in dimensions.keys():
+        configs = select_configs_for_platform(CONFIGS, dimensions["platform"])
+        assert_that(len(configs)).is_equal_to(dimensions["length"])
+    else:
+        configs = select_configs_for_scheduler(CONFIGS, dimensions["scheduler"])
+        assert_that(len(configs)).is_equal_to(dimensions["length"])
+
+
+@pytest.mark.parametrize(
+    "info",
+    [
+        {"node_info": {"dcv_enabled": "head_node", "directory_service": {"enabled": "true"}}, "length": 3},
+        {"node_info": {"directory_service": {"enabled": "true"}}, "length": 2},
+        {"node_info": {"enabled": "true"}, "length": 1},
+    ],
+)
+def test_select_configs_for_feature(mocker, info):
+    node_info = info["node_info"]
+    mocker.patch(
+        "write_cloudwatch_agent_json.get_node_info",
+        return_value=node_info,
+    )
+    selected_configs = select_configs_for_feature(CONFIGS)
+    assert_that(len(selected_configs)).is_equal_to(info["length"])
+
+
+def test_add_timestamps():
+    timestamp_formats = {"month_first": "%b %-d %H:%M:%S", "default": "%Y-%m-%d %H:%M:%S,%f"}
+    configs = add_timestamps(CONFIGS, timestamp_formats)
+    for config in configs:
+        timestamp_format = timestamp_formats[config["timestamp_format_key"]]
+        assert_that(config["timestamp_format"]).is_equal_to(timestamp_format)
+
+
+def test_filter_output_fields():
+    desired_keys = ["log_stream_name", "file_path", "timestamp_format", "log_group_name"]
+    configs = filter_output_fields(CONFIGS)
+    for config in configs:
```

Comment:
I am not sure if using part of the function code that you are testing in the test function itself is a good strategy.